Mathlib/GroupTheory/Coset/Basic.lean

```diff
@@ -402,6 +402,21 @@ theorem out_eq' (a : α ⧸ s) : mk a.out' = a :=
 
 variable (s)
 
+/-- Given a subgroup `s`, the function that sends a subgroup `t` to the pair consisting of
+its intersection with `s` and its image in the quotient `α ⧸ s` is strictly monotone, even though
+it is not injective in general. -/
+@[to_additive QuotientAddGroup.strictMono_comap_prod_image "Given an additive subgroup `s`,
+the function that sends an additive subgroup `t` to the pair consisting of
+its intersection with `s` and its image in the quotient `α ⧸ s`
+is strictly monotone, even though it is not injective in general."]
+theorem strictMono_comap_prod_image :
+    StrictMono fun t : Subgroup α ↦ (t.comap s.subtype, mk (s := s) '' t) := by
```

Comment:
The target type of this function is a mixed of Subgroup and Set, and I feel it would be better that it be homogeneous.

Reply:
otherwise, this is good to me.

Reply:
I could make the target `Set s × Set (α ⧸ s)` but I think the current version is potentially more useful, because e.g. if `s` is infinite then `Set s` isn't well-founded either way, but `Subgroup s` may well be.

Reply:
Why not making it `Subgroup s × Subgroup (α ⧸ s)` ?

Reply:
Why not making it `Subgroup s × Subgroup (α ⧸ s)` ?

Reply:
You can if `s` is normal, which is done in another file.

Reply:
Good.